M2/Macaulay2/m2/matrix.m2

```diff
@@ -127,6 +134,8 @@ Matrix + RingElement := (f,r) -> if r == 0 then f else f + r*id_(target f)
 RingElement + Matrix := (r,f) -> if r == 0 then f else r*id_(target f) + f
 Number + Matrix := (i,f) -> if i === 0 then f else i*id_(target f) + f
 Matrix + Number := (f,i) -> if i === 0 then f else f + i*id_(target f)
+Vector + Number := Vector + RingElement := (v,r) -> vector(matrix v + r)
+Number + Vector := RingElement + Vector := (r,v) -> vector(r + matrix v)
```

Comment:
```m2
i1 : Number + Vector := RingElement + Vector := (r,v) -> vector(r + matrix v);

i2 : 1 + (coker matrix {{ 10 }})_0
stdio:1:58:(3):[1]: error: expected source to be free with rank 1
```

Reply:
This is coming from `vector(Matrix)`:

```m2
i1 : code (vector, Matrix)

o1 = -- code for method: vector(Matrix)
     M2/Macaulay2/m2/modules.m2:66:17-69:5: --source code:
     vector Matrix := f -> (
          if not isFreeModule source f or numgens source f =!= 1 then error "expected source to be free with rank 1";
          new target f from {f}
         )
```

Should `Vector` objects be elements of free modules?  Or arbitrary modules?

Reply:
That's not the problem. Single generators of any module whatsoever are vectors. The implementation of `Number + Vector` isn't correct because it uses the identity matrix on the target.

Reply:
Hmm, this seems to be a problem with matrix addition:

```m2
i1 : M = coker matrix 10;

i2 : A = map(M, ZZ^1, 1)

o2 = | 1 |

                    1
o2 : Matrix M <-- ZZ

i3 : 1 + A

o3 = | 2 |

o3 : Matrix M <-- M

i4 : A + 1

o4 = | 2 |

                    1
o4 : Matrix M <-- ZZ
```

Shouldn't they both have `ZZ^1` as their source?

Reply:
I just pushed a new commit that should fix this